src/jit/codegencommon.cpp

```diff
@@ -10602,13 +10688,13 @@ void CodeGen::genSetScopeInfoUsingsiScope()
 // Notes:
 //    Called for every scope info piece to record by the main genSetScopeInfo()
 
-void CodeGen::genSetScopeInfo(unsigned       which,
-                              UNATIVE_OFFSET startOffs,
-                              UNATIVE_OFFSET length,
-                              unsigned       varNum,
-                              unsigned       LVnum,
-                              bool           avail,
-                              siVarLoc*      varLoc)
+void CodeGen::genSetScopeInfo(unsigned        which,
+                              UNATIVE_OFFSET  startOffs,
+                              UNATIVE_OFFSET  length,
+                              unsigned        varNum,
+                              unsigned        LVnum,
+                              bool            avail,
+                              const siVarLoc* varLoc)
```

Comment:
Adding this const is unnecessary, it will just force us to add an unnecessary copy of the type and it will not preserve the modified state to the caller.

Reply:
I agree with you specially in this one. It make sense to preserve the modifications of siVarLoc on the VariableLiveRanges. On the other side, this is the last use of VariableLiveRanges, after setting the array eeVars, we dump and set eeVars to the debugger. The bad side of passing a non-const reference/pointer here is that it makes getLiveRanges() return a non-const pointer too, which lets anyone modify the internal struct of VariableLiveDescriptor.